vault/request_handling.go

```diff
@@ -2287,10 +2316,30 @@ func (c *Core) RegisterAuth(ctx context.Context, tokenTTL time.Duration, path st
 		if te.ExternalID != "" {
 			auth.ClientToken = te.ExternalID
 		}
-		// Successful login, remove any entry from userFailedLoginInfo map
-		// if it exists. This is done for service tokens (for oss) here.
-		// For ent it is taken care by registerAuth RPC calls.
-		if userLockoutInfo != nil {
+
+		if c.standby.Load() {
+			// If we are a standby, notify the primary about the successful login
+			// The primary will use this to reset the user lockout counter
+
+			c.requestForwardingConnectionLock.RLock()
+			rpcForwardingClient := c.rpcForwardingClient
+			c.requestForwardingConnectionLock.RUnlock()
+
+			if rpcForwardingClient != nil {
+				c.logger.Trace("forwarding successful login attempt", "mount_accessor", userLockoutInfo.mountAccessor, "user_alias", userLockoutInfo.aliasName)
+				_, err = c.rpcForwardingClient.ForwardLoginAttempt(ctx, &forwarding.LoginAttempt{
+					NamespaceUuid: ns.UUID,
+					MountAccessor: userLockoutInfo.mountAccessor,
+					UserAliasName: userLockoutInfo.aliasName,
+					Successful:    true,
+				})
+				if err != nil {
+					c.logger.Warn("failed to forward successful login attempt", "error", err, "mount_accessor", userLockoutInfo.mountAccessor, "user_alias", userLockoutInfo.aliasName)
+				}
+			}
+		} else if userLockoutInfo != nil {
```

Comment:
```suggestion
		}

		if userLockoutInfo != nil {
```

Shouldn't these be separate?